dace/codegen/targets/cpp.py

```diff
@@ -50,24 +50,28 @@ def copy_expr(
             s = offset
         o = None
     if s is not None:
-        offset_cppstr = cpp_offset_expr(datadesc, s, o)
+        offset_cppstr = cpp_offset_expr(data_desc, s, o)
     else:
         offset_cppstr = "0"
     dt = ""
 
-    expr = ptr(dataname, datadesc)
+    def_type, _ = dispatcher.defined_vars.get(data_name)
 
-    def_type, _ = dispatcher.defined_vars.get(dataname)
+    expr = ptr(
+        data_name,
+        data_desc,
+        hbm_bank,
```

Comment:
To be honest I would not know how to do that easily. data_name and data_desc are definitively required as well as sdfg (one could save the data_desc by requiring sdfg to be always set). hbm_bank info is used as from the outside and changes often between calls. The actual reason why is_write, dispatcher, is_array_interface, ancestor and interface_id are there as well is because ptr was merged with array_interface_variable function (beeing seperate functions lead to problems since suddenly the order of calls to ptr and array_interface_variable matters, and even more because it's not so clear why ptr and array_interface_variable should be seperate functions, since they do the same thing in principle - generate the final name for a variable). It was allowed to call array_interface_variable when the variable was not yet defined, so dispatcher may be None, even if is_array_interface is True. Similarly one may have is_array_interface set to True with or without setting interface_id, they are independent. 
Guess the only way to reduce parameters I see would be to either remove is_array_interface, is_write and dispatcher and ancestor and expect some new parameter array_interface_info or so that is a tuple of is_write, dispatcher and ancestor (feels a bit forced), or to accept that ptr gets to complicated when merged with array_interface_variable and to split them again.

Reply:
I readded the array_interface_variable function

Reply:
Final solution is an fpga specific ptr in fpga_utils, that is called whenever we are dealing with fpga arrays. It's not part of the fpga-codegen because that's annoying/ugly to implement in cpp.